cpp/src/arrow/compute/key_hash.cc

```diff
@@ -471,7 +473,31 @@ Status Hashing32::HashBatch(const ExecBatch& key_batch, uint32_t* hashes,
 
   LightContext ctx;
   ctx.hardware_flags = hardware_flags;
-  ctx.stack = temp_stack;
+
+  const int64_t alloc_entry_length = column_arrays[0].length();
+  auto estimate_size = [&] {
+    // An estimate TempVectorStack usage size for Hashing32::HashMultiColumm.
+    const int64_t alloc_size1 =
+        2 * (alloc_entry_length * sizeof(uint32_t) + util::TempVectorStack::meta_size());
```

Comment:
Is this for `hash_temp_buf` and `null_hash_temp_buf`?

Reply:
Yes